src/main/java/com/arpnetworking/tsdcore/sinks/HttpSinkActor.java

```diff
@@ -270,42 +338,62 @@ private void dispatchPending() {
+        }
 
         final Request request = requestEntry.getRequest();
         _inflightRequestsCount++;
 
+        fireRequest(request, 1);
+    }
+
+
+    private void fireRequest(final Request request, final int attempt) {
         final CompletableFuture<Response> promise = new CompletableFuture<>();
-        metrics.startTimer(_requestLatencyName);
+        final long requestStartTime = System.currentTimeMillis();
         _client.executeRequest(request, new ResponseAsyncCompletionHandler(promise));
         final CompletionStage<Object> responsePromise = promise
                 .handle((result, err) -> {
-                        metrics.stopTimer(_requestLatencyName);
-                        final Object returnValue;
-                        if (err == null) {
-                            final int responseStatusCode = result.getStatusCode();
-                            final int responseStatusClass = responseStatusCode / 100;
-                            for (final int i : STATUS_CLASSES) {
-                                metrics.incrementCounter(
-                                        String.format("%s/%dxx", _responseStatusName, i),
-                                        responseStatusClass == i ? 1 : 0);
-                            }
-                            if (ACCEPTED_STATUS_CODES.contains(responseStatusCode)) {
-                                 returnValue = new PostSuccess(result);
-                            } else {
-                                 returnValue = new PostRejected(request, result);
-                            }
+                    try (Metrics metrics = _metricsFactory.create()) {
+                        metrics.setTimer(_requestLatencyName, System.currentTimeMillis() - requestStartTime, TimeUnit.MILLISECONDS);
+                    }
+                    if (err == null) {
+                        if (ACCEPTED_STATUS_CODES.contains(result.getStatusCode())) {
+                             return new PostSuccess(attempt, result);
                         } else {
-                            returnValue = new PostFailure(request, err);
+                             return new PostRejected(attempt, request, result);
                         }
-                        metrics.incrementCounter(_requestSuccessName, (returnValue instanceof PostSuccess) ? 1 : 0);
-                        metrics.close();
-                        return returnValue;
+                    } else {
+                        return new PostFailure(attempt, request, err);
+                    }
                 });
         PatternsCS.pipe(responsePromise, context().dispatcher()).to(self());
     }
 
+    private void scheduleRetry(final Request request, final int attempt) {
+        final Duration delay = Duration.ofMillis(
+                Math.min(
+                        _sink.getBaseBackoff().getMillis() * ThreadLocalRandom.current().nextInt((int) Math.pow(2, attempt)),
```

Comment:
So the delayed between the first and second attempt will have a delay computed as: `base * nextInt(2 ^ 1)`; because the first attempt is fired with `fireRequest(request, 1);` and it's only incremented on retry (after handling of the reject/failure). 

Is this what we want? 

e.g. for base is 50ms, delays are up to 100, 200, 400, 800, etc.

Are we off by one? (e.g. should it be 50, 100, 200, 400, 800, etc.)